content/blog/2021-11-17-exp-versioning.md

```diff
@@ -0,0 +1,243 @@
+  - MLOps
+  - DVC
+  - Experiments
+  - Experiment Tracking
+  - Experiment Versioning
+---
+
+Traditional software version control is not enough for tracking machine learning
+experiments. Over the last few years, experiment tracking has come a long way,
+but there are parallels to problems from earlier generations of version control.
+That's why "experiment versioning" in DVC tracks experiments by building on
+modern version control practices.
+
+# Experiment Tracking Needs
+
+When tracking ML experiments, you need to:
+
+- Log experiment information.
+- Organize and compare experiments.
+- Reproduce experiments.
+
+Tracking data, comparing versions, and enabling reproducibility are already
+addressed by version control. Unfortunately, traditional version control tools
+like Git don't work well for ML experimentation because:
+
+- Data and models aren't easily tracked with Git.
+- There are too many experiments to track with any branching workflow.
+- Machine learning pipelines take too long to reproduce on the fly.
+
+# History of Experiment Tracking
+
+Back in the old days of the 2010's, you would have had to track all this
+yourself, using a notebook, spreadsheet, or some custom solution. This limits
+you to scalar parameters and metrics, and leaves a lot of room for human error
+(forgetting to log a particular value or a whole experiment, or writing down the
+wrong value). Organizing and comparing experiments was limited to sorting
+spreadsheets. Reproducing experiments was still left to traditional version
+control, if that was even used.
+
+Experiment tracking tools were developed to better address experiment tracking
+needs. They provide you with an API to log experiment information, a database to
+store it, and a dashboard to compare and visualize. You can log all types of
+artifacts, label and compare experiments in a dashboard, and collect logged
+artifacts to reproduce the experiment.
+
+However, as projects grow, these experiment tracking tools leave a lot up to you
+and start to feel cumbersome:
+
+- You need to decide what is important to log. Much of the information is the
+  same between experiments, and it's not clear what changed in each.
+- You need to keep experiments organized, you need to manage who has access to
+  them, and you may need to drop experiments that are outdated or otherwise
+  irrelevant.
+- You need to set up a process to reproduce an experiment from the logged
+  artifacts, running it again from scratch. If any input is not logged,
+  reproducibility is lost.
+
+# Experiment Versioning in DVC
+
+How does DVC help?
+
+- Log experiments end to end (commands, inputs, and outputs) and track changes
+  introduced by each experiment.
+- Associate experiments with Git commits, branches, and tags from which they
+  derive. Organize locally and choose what to share.
+- Reproduce any previous experiment with a single command. Cache previous runs
+  and results to reduce duplicate compute time and storage space.
```

Comment:
This may be me, but it would help to have a simple example of the difference. Let's say. Here's a problem. When we solve it with some other tool (not mentioning), this is what we can log and track. 
When we solve it with DVC, these are the additional benefits.